### PR TITLE
feat(FX-3274): fallback name for saved search alert without name

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3274]

### Demo
iPhone 8 – 14 4 2021-09-02 19-30-41

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Display fallback name for saved search alert without name - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3274]: https://artsyproduct.atlassian.net/browse/FX-3274

### CODE REVIEW

File: src/lib/Scenes/SavedSearchAlert/Components/Form.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ export const Form: React.FC<FormProps> = (props) => {
       <Spacer mt={4} />
       <Button
         testID="save-alert-button"
-        disabled={!!savedSearchAlertId && !dirty}
+        disabled={!!savedSearchAlertId && !(dirty || values.name.length === 0)}
```
Comment (dimatretyak) [inline src/lib/Scenes/SavedSearchAlert/Components/Form.tsx:78]: Should the "save alert" button be enabled if the user opens the edit saved search alert page without saved name (by default, the button is disabled for the update form until the user has changed the alert name)?

https://user-images.githubusercontent.com/3513494/131888660-4332e94e-e737-4d54-856c-772fa31fcc83.mp4

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Display fallback name for saved search alert without name - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 748a75f232222ed03198208b5f094c7196e02b63
</p>

File: src/lib/Scenes/SavedSearchAlert/Components/Form.tsx (hunk 1 above)
Comment (mdole) [reply src/lib/Scenes/SavedSearchAlert/Components/Form.tsx:78]: Good question! What you demonstrated in the gif is the correct behavior - a user should be able to save right away if they do not have a name already.

**Review by mdole — APPROVED** [review]
Nice work @dimatretyak! Let's ship it!

**Pull request merged**